src/hyperactive/integrations/skforecast/skforecast_opt_cv.py

```diff
@@ -0,0 +1,200 @@
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator.
+
+        Parameters
+        ----------
+        parameter_set : str, default="default"
+            Name of the parameter set to return.
+
+        Returns
+        -------
+        params : dict or list of dict, default = {}
+            Parameters to create testing instances of the class
+            Each dict are parameters to construct an "interesting" test instance,
+            i.e., MyClass(**params) or MyClass(**params[i]) creates a valid test
+            instance.
+            create_test_instance uses the first (or only) dictionary in `params`
+        """
+        from skforecast.recursive import ForecasterRecursive
+        from sklearn.ensemble import RandomForestRegressor
+
+        from hyperactive import HillClimbingOptimizer
+
+        forecaster = ForecasterRecursive(
+            regressor=RandomForestRegressor(random_state=123),
+            lags=2,
+        )
+        optimizer = HillClimbingOptimizer()
+
+        params = {
+            "forecaster": forecaster,
+            "optimizer": optimizer,
+            "steps": 3,
+            "metric": "mean_squared_error",
+            "initial_train_size": 10,
+        }
+        return [params]
+
+    def fit(self, y, exog=None):
+        """Fit to training data.
+
+        Parameters
+        ----------
+        y : pandas Series
+            Target time series to which to fit the forecaster.
+        exog : pandas Series or DataFrame, optional
+            Exogenous variables.
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        current_exog = exog if exog is not None else self.exog
+
+        experiment = SkforecastExperiment(
+            forecaster=self.forecaster,
+            y=y,
+            steps=self.steps,
+            metric=self.metric,
+            initial_train_size=self.initial_train_size,
+            exog=current_exog,
+            refit=self.refit,
+            fixed_train_size=self.fixed_train_size,
+            gap=self.gap,
+            allow_incomplete_fold=self.allow_incomplete_fold,
+            return_best=self.return_best,
+            n_jobs=self.n_jobs,
+            verbose=self.verbose,
+            show_progress=self.show_progress,
+        )
+
+        if hasattr(self.optimizer, "clone"):
+            optimizer = self.optimizer.clone()
+        else:
+            optimizer = copy.deepcopy(self.optimizer)
+
+        optimizer.set_params(experiment=experiment)
+        best_params = optimizer.solve()
+
+        self.best_params_ = best_params
+        self.best_forecaster_ = copy.deepcopy(self.forecaster)
+        self.best_forecaster_.set_params(best_params)
+
+        # Refit model with best parameters on the whole dataset
+        self.best_forecaster_.fit(y=y, exog=current_exog)
+
+        return self
+
+    def predict(self, steps, exog=None, **kwargs):
```

Comment:
I will also include the `last_window` parameter in the `predict` method. You can copy the docstring from here:

https://skforecast.org/0.18.0/api/forecasterrecursive#skforecast.recursive._forecaster_recursive.ForecasterRecursive.predict

`check_inputs` is for internal use only, so it is not worth including it.